dm/dm/common/common.go

```diff
@@ -215,6 +216,19 @@ var (
 	// StageRelayKeyAdapterV1 is used to store the running stage of the relay.
 	// k/v: Encode(source-id) -> the running stage of the relay.
 	StageRelayKeyAdapterV1 KeyAdapter = keyEncoderDecoder("/dm-master/stage/relay/")
+	// TODO: support rolling upgrade from lower versions for UpstreamBoundWorkerKeyAdapterV1, UpstreamLastBoundWorkerKeyAdapterV1, UpstreamRelayWorkerKeyAdapterV1
```

Comment:
also remeber delete these keys

Reply:
Do you mean `UpstreamBoundWorkerKeyAdapterV1`? These keys need to be kept because we should handle the situation that we upgrade from lower versions.

Reply:
ok, ok but this may result in these keys never being deleted, maybe provide a scripct or command for user to delete those old key

Reply:
Yes. We should delete these keys once we successfully upgrade to a newer version.